pyproject.toml

```diff
@@ -30,37 +33,106 @@ classifiers = [
     "Topic :: Utilities",
     "Typing :: Typed",
 ]
-dynamic = ["version", "dependencies", "optional-dependencies"]
+dynamic = ["version"]
 
 [project.urls]
 Changelog = "https://docs.disnake.dev/page/whats_new.html"
 Documentation = "https://docs.disnake.dev/"
 Repository = "https://github.com/DisnakeDev/disnake"
 
+[project.optional-dependencies]
+speed = [
+    "orjson~=3.6",
+    # taken from aiohttp[speedups]
+    "aiodns>=1.1",
+    "Brotli",
+    'cchardet; python_version < "3.10"',
+]
+voice = [
+    "PyNaCl>=1.3.0,<1.5",
+]
+docs = [
+    "sphinx==6.0.1",
+    "sphinxcontrib-trio~=1.1.2",
+    "sphinx-hoverxref==1.3.0",
+    "sphinx-autobuild~=2021.3",
+    "sphinxcontrib-towncrier==0.3.2a0",
+    "towncrier==22.12.0",
+    "sphinx-notfound-page==0.8.3",
+]
+discord = ["discord-disnake"]
+
+[tool.pdm.dev-dependencies]
+tools = [
+    "pre-commit~=2.19.0",
+    "taskipy~=1.10.1",
+    "slotscheck~=0.15.0",
+    "python-dotenv~=0.20.0",
+    "towncrier==22.12.0",
+    "check-manifest==0.48",
+]
+codemod = [
+    # run codemods on the respository (mostly automated typing)
+    "libcst~=0.4.7",
+    "black==23.1.0",
+]
+lint = [
+    "flake8~=4.0.1",
+    "flake8-bandit~=3.0.0",
+    "flake8-bugbear==22.7.1",
+    "flake8-comprehensions~=3.10.0",
+    "flake8-noqa~=1.2.5",
+    "flake8-pytest-style~=1.6.0",
+]
+typing = [
+    "typing-extensions~=4.2.0",
+    # this is not pyright itself, but the python wrapper
+    "pyright==1.1.291",
+]
+test = [
+    "pytest~=7.2.0",
+    "pytest-cov~=4.0.0",
+    "pytest-asyncio~=0.20.3",
+    "looptime~=0.2",
+    "coverage[toml]~=6.5.0",
+]
+
+[tool.pdm.scripts]
+black = { composite = ["lint black"], help = "Run black" }
+docs = { cmd = "nox -Rs docs --", help = "Build the documentation for development" }
+flake8 = { cmd = "flake8", help = "Run flake8" }
+isort = { composite = ["lint isort"], help = "Run isort" }
+lint = { cmd = "nox -Rs lint --", help = "Check all files for linting errors" }
+pyright = { cmd = "nox -Rs pyright --", help = "Run pyright" }
+setup_env = { cmd = "pdm install -d -G speed -G docs -G voice", help = "Setup the local environment and set up all dependencies" }
+post_setup_env = "pre-commit install --install-hooks"
```

Comment:
could install pip with a composite `post_setup_env` here, but it's not *really* needed.

Reply:
```suggestion
setup_env = { cmd = "pdm install -d -G speed -G docs -G voice", help = "Setup the local environment and set up all dependencies" }
post_setup_env = { composite = ["python -m ensurepip --default-pip", "pre-commit install --install-hooks"] }
```

Reply:
https://github.com/DisnakeDev/disnake/pull/836/commits/1b49869aa1a7496a7a36a9a4f27c31252e541d60